### PR TITLE
dcm2niix 1.0.20170609 (new formula)

### PR DESCRIPTION
**Pull request by dangom**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Add dcm2niix, a Dicom to Nifti conversion tool.

### CODE REVIEW

File: Formula/dcm2niix.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Dcm2niix < Formula
+  desc "DICOM to NIfTI converter"
+  homepage "https://www.nitrc.org/plugins/mwiki/index.php/dcm2nii:MainPage"
+  url "https://github.com/rordenlab/dcm2niix/archive/v1.0.20170609.tar.gz"
+  sha256 "17756f28dc42965854b7389d166df6d70c126486887fd2037b99d4a082c037ab"
+  head "https://github.com/rordenlab/dcm2niix.git"
+
+  option "with-dcm2niibatch"
```
Comment (dunn) [inline Formula/dcm2niix.rb]: Could this just be `"with-batch"`?

File: Formula/dcm2niix.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  depends_on "cmake" => :build
+
+  def install
+    if build.with? "dcm2niibatch"
```
Comment (dunn) [inline Formula/dcm2niix.rb]: For extra options, instead we usually do:
```ruby
args = std_cmake_args
args << "-DBATCH_VERSION=ON" if build.with? "batch"
system "cmake", ".", *args

File: Formula/dcm2niix.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      system "cmake", ".", "-DBATCH_VERSION=ON", *std_cmake_args
+    else
+      system "cmake", ".", *std_cmake_args
+    end
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/dcm2niix"
```
Comment (dunn) [inline Formula/dcm2niix.rb]: Does this just display the help text? It would be good to have a test that checks more of the functionality.

**Review by dunn — CHANGES_REQUESTED** [review]

File: Formula/dcm2niix.rb (hunk 3 above)
Comment (dangom) [inline Formula/dcm2niix.rb]: This just displays the help text.

I thought about having better tests. The problem is that dcm2niix is an image converter, which means it needs a dicom image to exist to really test the functionality. Is there a way of downloading a file and converting it, then removing it? Would that be safe?

File: Formula/dcm2niix.rb (hunk 2 above)
Comment (dangom) [inline Formula/dcm2niix.rb]: looks nice. I'll update it.

File: Formula/dcm2niix.rb (hunk 1 above)
Comment (dangom) [inline Formula/dcm2niix.rb]: Yes, definitely.

File: Formula/dcm2niix.rb (hunk 3 above)
Comment (dunn) [inline Formula/dcm2niix.rb]: Definitely! You can use a `resource` block to fetch an image; the test is run in a temporary directory so you don't need to worry about cleaning up manually.  [`potrace.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/potrace.rb) and [`blockhash.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/blockhash.rb) are two examples; you can grep around the code for more.

Comment (dangom) [inline Formula/dcm2niix.rb]: Wow, now that was easy. =) Test updated to check that all relevant functionality works.

**Review by dunn — APPROVED** [review]

**Comment by dunn** [comment]
Looks great, thanks for contributing! Merged it as 22f5c2ba1a9a666b9943da8ba97bf4420ac40f75.

**Pull request closed without merging**